lib/administrate/order.rb

```diff
@@ -68,7 +73,30 @@ def order_by_count(relation)
     end
 
     def order_by_id(relation)
-      relation.reorder("#{foreign_key(relation)} #{direction}")
+      relation.reorder(Arel.sql(order_by_id_query(relation)))
+    end
+
+    def order_by_attribute(relation)
+      return order_by_id(relation) unless field
+
+      relation.joins(
+        attribute.to_sym,
```

Comment:
Is that required? It'll remove the records with no value in the specified column.